features/lorawan/lorastack/mac/LoRaMac.h

```diff
@@ -453,10 +453,24 @@ class LoRaMac {
     lorawan_status_t clear_tx_pipe(void);
 
     /**
-     * Gets the current time
+     * Gets the current monotonic tick time
      */
     lorawan_time_t get_current_time(void);
 
+    /**
+     * Gets the current GPS time
+     * For precise information please refer to LoRaWANTimer::get_gps_time() API
+     * documentation.
+     */
+    lorawan_time_t get_gps_time(void);
```

Comment:
Woah there - you're using `lorawan_time_t` for two totally different timebases here. I'm not even sure they're running at the same rate, and they certainly don't have the same epoch. Seems like you need a new type for this.

Or is it 3 different timebases? Monotonic milliseconds, GPS milliseconds and GPS seconds?